app/reducers/reducer_pins.js

```diff
@@ -22,8 +23,18 @@ export default function(state = initialState, action) {
       return Object.assign({}, state,
         newPin
       );
+    case DELETE_PIN:
+      let id = action.payload.id
+      let deletedPinState = Object.assign({}, state);
+      deletedPinState[id] = Object.assign({}, state[id]);
+      delete deletedPinState[id];      
+
+      return Object.assign({},
```

Comment:
I believe that will delete the pin that both the state and newState are referencing